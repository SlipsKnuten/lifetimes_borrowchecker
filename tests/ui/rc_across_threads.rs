use std::{rc::Rc, thread};

fn main() {
    let r = Rc::new(5);
    thread::spawn(move || {
        let _ = r.clone();
    });
}
