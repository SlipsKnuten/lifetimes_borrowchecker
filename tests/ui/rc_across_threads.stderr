error[E0277]: `Rc<i32>` cannot be sent between threads safely
 --> tests/ui/rc_across_threads.rs:5:19
  |
5 |       thread::spawn(move || {
  |       ------------- ^------
  |       |             |
  |  _____|_____________within this `{closure@$DIR/tests/ui/rc_across_threads.rs:5:19: 5:26}`
  | |     |
  | |     required by a bound introduced by this call
6 | |         let _ = r.clone();
7 | |     });
  | |_____^ `Rc<i32>` cannot be sent between threads safely
  |
  = help: within `{closure@$DIR/tests/ui/rc_across_threads.rs:5:19: 5:26}`, the trait `Send` is not implemented for `Rc<i32>`
note: required because it's used within this closure
 --> tests/ui/rc_across_threads.rs:5:19
  |
5 |     thread::spawn(move || {
  |                   ^^^^^^^
note: required by a bound in `spawn`
 --> $RUST/std/src/thread/mod.rs
